src/fillTank.test.js

```diff
@@ -1,11 +1,99 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared ', () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it('should calculate full tank If the amount is not given', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 0,
+      },
+    };
+
+    fillTank(customer, 30);
+    expect(customer.money).toBe(1800);
+    expect(customer.vehicle.fuelRemains).toBe(40);
+  });
+
+  it(
+    'should calculate full tank If the amount is greater than the tank can use',
+    () => {
+      const customer = {
+        money: 3000,
+        vehicle: {
+          maxTankCapacity: 20,
+          fuelRemains: 5,
+        },
+      };
+
+      fillTank(customer, 50, 20);
+
+      expect(customer.money).toBe(2250);
+      expect(customer.vehicle.fuelRemains).toBe(20);
+    });
+
+  it('should fill the fuel if the client can pay for it', () => {
+    const customer = {
+      money: 200,
+      vehicle: {
+        maxTankCapacity: 30,
+        fuelRemains: 5,
+      },
+    };
+
+    fillTank(customer, 20, 20);
+
+    expect(customer.money).toBe(0);
+    expect(customer.vehicle.fuelRemains).toBe(15);
+  });
+
+  it('should round the amount by discarding number to the tenth part.', () => {
+    const customer = {
+      money: 600,
+      vehicle: {
+        maxTankCapacity: 50,
+        fuelRemains: 5,
+      },
+    };
+
+    fillTank(customer, 9, 10.239);
```

Comment:
Ensure that the `fillTank` function rounds the poured amount by discarding numbers to the tenth part. The expected fuel remains should be checked for correct rounding.